mod-source-record-storage-server/src/main/java/org/folio/services/util/AdditionalFieldsUtil.java

```diff
@@ -372,14 +370,25 @@ public static void remove003FieldIfNeeded(Record record, String instanceHrid) {
 
   /**
    * Check if record should be filled by specific fields.
-   * @param record - source record.
-   * @param instance - source instance.
+   *
+   * @param record         - source record.
+   * @param externalEntity - source externalEntity.
    * @return - true if need.
    */
-  public static boolean ifFillingFieldsNeeded(Record record, JsonObject instance) {
-    return (StringUtils.isNotEmpty(record.getExternalIdsHolder().getInstanceId()) && StringUtils.isNotEmpty(record.getExternalIdsHolder().getInstanceHrid())) &&
-      (instance.getString(ID_FIELD).equals(record.getExternalIdsHolder().getInstanceId())
-        && !instance.getString(HR_ID_FIELD).equals(record.getExternalIdsHolder().getInstanceHrid()));
+  public static boolean ifFillingFieldsNeeded(Record record, JsonObject externalEntity) {
```

Comment:
could you rename method to isFieldsFillingNeeded to align with already existing method names such as isNotEmpty(), isEligible(), isPresent()

Reply:
Renamed